compose/testcontainers/compose/__init__.py

```diff
@@ -65,14 +70,33 @@ def __enter__(self) -> "DockerCompose":
     def __exit__(self, exc_type, exc_val, exc_tb) -> None:
         self.stop()
 
+    def base_docker_compose(self):
+        """
+        Returns the basecommand parts used for the docker compose commands
+        depending on the docker compose api.
+
+        Returns
+        -------
+        list[str]
+            The docker compose command parts
+        """
+        if self._user_defined_compose_command:
+            return self._user_defined_compose_command
+
+        if subprocess.run(["docker", "compose", "--help"], stdout=subprocess.DEVNULL,
+                          stderr=subprocess.STDOUT).returncode == 0:
+            return ["docker", "compose"]
+
+        return ["docker-compose"]
```

Comment:
can you run this once on the constructor? it would save some noise and confusion and it's better to fail early.

Reply:
good idea!